tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
-        suite = junit.make_junit_xml(self)
-        data = junit_xml.TestSuite.to_xml_string([suite])
-
-        # Zip the report
-        bytestream = io.BytesIO()
-        # SIM117: Use a single `with` statement with multiple contexts instead of nested `with`
-        # statements. Here, `zipstream` is being used in the second nested context.
-        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,  # noqa: SIM117
-                             compresslevel=1) as zipstream:
-            # XML file names are irrelevant to ReportPortal
-            with zipstream.open("tests.xml", "w") as entry:
-                entry.write(data.encode("utf-8"))
-        bytestream.seek(0)
-
-        # Send the report to the ReportPortal instance
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
+        url = f"{server}/api/{version}/{project}"
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        tmt_vars = self.get("tmt-vars")
+        attributes = [{'key': k, 'value': v[0]} for k, v in self.step.plan._fmf_context.items()]
+
+        # Communication with RP instance
         with tmt.utils.retry_session() as session:
-            url = f"{server}/api/v1/{project}/launch/import"
-            self.debug(f"Send the report to '{url}'.")
-            response = session.post(
-                url,
-                headers={
-                    "Authorization": "bearer " + token,
-                    "accept": "*/*",
-                    },
-                files={
-                    # The zip filename is used as the launch name in ReportPortal
-                    "file": (launch_name + ".zip", bytestream, "application/zip"),
-                    },
-                )
-
-        # Handle the response
-        try:
-            message = tmt.utils.yaml_to_dict(response.text).get("message")
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
 
-        self.debug(f"Response code from the server: {response.status_code}")
-        self.debug(f"Message from the server: {message}")
-        self.info("report", "Successfully uploaded.", "yellow")
+            # Create a launch
+            self.info("launch", launch_name, color="cyan")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
```

Comment:
Hmm, I suppose you are aware this timestamp is actually not the time when `tmt run ...` started. Unfortunately, tmt does not have any attribute that would provide you with the correct date/time.

But, we do have a start/end time for every *result* - I would recommend using it instead of `time.time()` which is not related to actual test timestamps. See `Result.starttime` and `Result.endtime`, it should be easy to convert them to a timestamp the API apparently prefers. I'd recommend using the `endtime` for `time` of the logs and failures code submits below as well, it would be much better reflecting the reality.

Reply:
Yes, you're right..
I did not really focus on the time of test execution but on actual report to RP.
I'll try to rewrite it as you recommend, although the times will be repeated quite a lot:
- LAUNCH start_time ..  ??  `result[0].starttime` ??
-  ITEM start_time ........ `result.starttime`
- LOG - ERROR time .... `result.endtime`
- LOG - INFO time ....... `result.endtime`
- LOGs - TRACE time ... `result.endtime`
- ITEM end_time ......... `result.endtime`
- LAUNCH end_time ...  ??  `timestamp()` ??  `result[n].endtime` ??
Is this, what you suggest?
What do you recommend to use for start of the launch? Result.starttime of the first result or is there any other value that can be used?
Same for the end of the launch? Result.endtime of the last result or leave there an actual timestamp when reporting the end of the launch?

Reply:
> Yes, you're right.. I did not really focus on the time of test execution but on actual report to RP.

Yep, that's fine, np.

> I'll try to rewrite it as you recommend, although the times will be repeated quite a lot:
> 
> * LAUNCH start_time ..  ??  `result[0].starttime` ??

Probably the best we have right now, although please, file an issue, tmt should save start/end times for run and plan as well, somewhere, somehow.

> * ITEM start_time ........ `result.starttime`
> * LOG - ERROR time .... `result.endtime`
> * LOG - INFO time ....... `result.endtime`
> * LOGs - TRACE time ... `result.endtime`
> * ITEM end_time ......... `result.endtime`

All these are fine, tmt does not really track when each log was created (maybe `stat` and `ctime` would be usable, but also, maybe not as they could be fetched from the guest), but `endtime` would look good to me, a solid beginning. One can always treat it as if the log became available when the test finished, which is tru.

> * LAUNCH end_time ...  ??  `timestamp()` ??  `result[n].endtime` ??

Same as `starttime` - good enough to start with, but mention it in the issue and let's provide a better one.

Reply:
Okay, thanks for help.
Already updated! Luckilly, it didn't require to convert the format.
And also submitted a new issue [#2180](https://github.com/teemtee/tmt/issues/2180) as suggested.